pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -95,6 +99,7 @@
   private volatile long _minTime = Long.MAX_VALUE;
   private volatile long _maxTime = Long.MIN_VALUE;
   private final int _numKeyColumns;
+  private final Collection<FieldSpec> _physicalColumnFieldSpecs;
```

Comment:
Let's cache physical dimensions and physical metrics as well to eliminate all the record level check

Reply:
I am not convinced that this is worth the change. In some places we look for fieldspec, and in others the column names. 
Why don't we get this PR through, and if we find that the 'if' checks are expensive, we can work on other ways to reduce them

Reply:
For the places where we get the column name, we use the column name to fetch the field spec. Instead, we can directly cache the field spec.
These if checks won't cause big performance issue, but we are doing redundant work over and over again.